hydrus/hydraspec/openapi-hydra_parser.py

```diff
@@ -0,0 +1,113 @@
+    """
+    Get Definitions from Open Api specification and convert to Classes and supported Props for Hydra Documentation
+    """
+    for class_ in definitions:
+        try:
+            desc = definitions[class_]["description"]
+        except KeyError:
+            desc = class_
+        classDefinition = HydraClass(class_, class_, desc, endpoint=False)
+        properties = definitions[class_]["properties"]
+        classAndClassDefinition[class_] = classDefinition
+        definitionSet.add(class_)
+        api_doc.add_supported_class(classDefinition, collection=False)
+        for prop in properties:
+            new_prop = HydraClassProp("vocab:" + prop, prop, required=False, read=True, write=True)
+            classAndClassDefinition[class_].add_supported_prop(new_prop)
+
+    get_ops()
+
+
+def generateEntrypoint():
+    """
+    Generates Entrypoint , Base Collection and Base Resource for the documentation
+    """
+    api_doc.add_baseCollection()
+    api_doc.add_baseResource()
+    api_doc.gen_EntryPoint()
+
+
+def get_ops():
+    """
+    Parses Paths from the Open Api Spec and creates supported operations for the paths which we have defined classes for
+    we check if the path has a class defined , if it does we parse the methods and add the information parsed to Hydra
+
+    :return:
+    """
+    for path in paths:
+        possiblePath = path.split('/')[1]
+        # dirty hack , do  case insensitive search more gracefully
+        possiblePath = possiblePath.replace(possiblePath[0], possiblePath[0].upper())
+        # check if the path name exists in the classes defined
+        if possiblePath in definitionSet and len(path.split('/')) == 2:
+            for method in paths[path]:
+                op_name = ""
+                op_method = method
+                op_expects = ""
+                op_returns = None
+                op_status = [{"statusCode": 200, "description": "dummyClass updated"}]
+                try:
+                    op_name = paths[path][method]["summary"]
+                except KeyError:
+                    op_name = possiblePath
+                try:
+                    parameters = paths[path][method]["parameters"]
+                    for param in parameters:
+                        op_expects = param["schema"]["$ref"].split('/')[2]
+                except KeyError:
+                    op_expects = None
+                classAndClassDefinition[possiblePath].add_supported_op(HydraClassOp(op_name,
+                                                                                    op_method,
+                                                                                    "vocab:" + op_expects,
+                                                                                    op_returns,
+                                                                                    op_status))
+                api_doc.add_supported_class(classAndClassDefinition[possiblePath], collection=False)
```

Comment:
Again, will only work for paths like `/User`. Also, what about collection type?

**Important - You're adding same class twice to the doc. First in line 41 and then again here.**
Why?

Reply:
there would be some classes which are only made to be referenced later on , classes are added when their parsing is done , later if we find operations on them we add the operations and add it to the api doc again , the class appears only once in the final doc . Please suggest a better way .